lib/omni_auth/strategies/abstract_saml.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require 'omniauth'
+require 'omniauth-saml'
+require 'ruby-saml'
+
+module OmniAuth
+  module Strategies
+    class AbstractSaml < OmniAuth::Strategies::SAML
+      # We want to specify some security options ourselves
+      option :security, {
+        digest_method: XMLSecurity::Document::SHA512,
+        signature_method: XMLSecurity::Document::RSA_SHA512,
+        metadata_signed: true, # Enable signature on Metadata
+        authn_requests_signed: true, # Enable signature on AuthNRequest
+        logout_requests_signed: true, # Enable signature on Logout Request
+        logout_responses_signed: true, # Enable signature on Logout Response
+        want_assertions_signed: true, # Require the IdP to sign its SAML Assertions
+        want_assertions_encrypted: true # Invalidate SAML messages without an EncryptedAssertion
+      }
+
+      # Our auto-login mechanism passes a desired redirect path (signed with a
+      # secret to prevent tampering) to the request phase.
+      # If we forward this via SAML's standard "RelayState" parameter, we will
+      # get it back in the callback phase.
+      # TODO: This parameter is not yet used in the app
```

Comment:
As we don't have any links to content from the start page (and we don't require redirecting the user after login for our prototype), I omitted the concrete implementation. The idea and option is taken from Xikolo.